### ISSUE TITLE
Fix all typescript warnings and errors generated by type mismatching on the `ConfigService.get(<envName>)` method in test specs

### ISSUE THREAD

**Issue by natanasow** [hashgraph/hedera-json-rpc-relay#3142]
##  Description
There are several places in current tests where expected values are in different types than those defined in the `ConfigService.get(<envName>)` method.

### Solution:
Identify all places where there are warnings or errors due to type mismatches. Remove any unnecessary casting, or add casting if required, to align with the types defined in `ConfigService.get(<envName>)`.

### Example Implementation
Refactor test cases to ensure type consistency with `ConfigService.get(<envName>)`, addressing any type warnings or errors.

**Comment by Mrinank-Bhowmick** [comment]
Hi! Could you please assign this issue to me? I'd love to work on it.

**Comment by Mrinank-Bhowmick** [comment]
After reviewing the requirements for this issue, I've realized I'm not able to take it on at this time. I'm unassigning myself to allow others to work on it.

**Comment by quiet-node** [comment]
Thanks @Mrinank-Bhowmick!

**Comment by belloibrahv** [comment]
@quiet-node I will like to work on this issue.

**Comment by belloibrahv** [comment]
Hi @quiet-node and @Nana-EC,  

I’d like to request permission to work on [[Issue #3142](https://github.com/hashgraph/hedera-json-rpc-relay/issues/3142)](https://github.com/hashgraph/hedera-json-rpc-relay/issues/3142), which focuses on fixing TypeScript warnings and errors related to type mismatches in `ConfigService.get(<envName>)`.  

I’ve reviewed the issue details and understand that it involves identifying and resolving type inconsistencies in the test specs by refactoring test cases for type alignment.  

Please let me know if I can take this on or if there are any specific guidelines you’d like me to follow while working on this.  

Looking forward to your feedback!  

Best regards,  
Ibrahim Bello

**Comment by quiet-node** [comment]
Hi @belloibrahv

I apologize for the delayed response; I completely overlooked this. Really appreciate your initiative! I've assigned you to the ticket.

I believe you're on the right track. Here’s a helpful reference for the solution: https://github.com/hashgraph/hedera-json-rpc-relay/issues/3142#issuecomment-2438213349

**Comment by belloibrahv** [comment]
OK, @quiet-node thanks for assigning the issue to me, currently looking into it 🚀

**Comment by belloibrahv** [comment]
Hi @quiet-node @natanasow, I noticed the 'NON_EXISTING_VAR' does not exist in the `GlobalConfig.ENTRIES` as part of the `envName` listed. I suggest adding it to ensure type safety when retrieving configuration values. Or do what you think about it.
<img width="904" alt="Screenshot 2024-12-12 at 14 38 41" src="https://github.com/user-attachments/assets/0b774c4a-e683-4539-9109-78001621b351" />

**Comment by belloibrahv** [comment]
Hi @quiet-node @natanasow,  

I hope this message finds you well. I’m currently working on the issue #3142 (`Fix all TypeScript warnings and errors generated by type mismatching on the ConfigService.get(<envName>) method in test specs`).  

While committing and signing off my changes, I encountered several linter errors during the build-and-test process. These errors seem unrelated to the code changes I’ve made, as they primarily point to existing issues in the repository. Below are a few examples of the errors:  

- **Error**: `'accounts' is never reassigned. Use 'const' instead` (`prefer-const`)  
- **Warning**: `Require statement not part of import statement` (`@typescript-eslint/no-var-requires`)  
- **Warning**: `'data' is defined but never used` (`@typescript-eslint/no-unused-vars`)  

These issues are originating from files such as:  
- `/tests/integration/server.spec.ts`  
- `/tests/acceptance/getTransactionByHash.spec.ts`  
- `/tests/acceptance/sendRawTransaction.spec.ts`, and others.  

I’ve ensured that my changes align with the task requirements, and the issues above were not introduced by my code.  

Could you kindly guide me on how to proceed? Should I address these linter issues as part of this task, or would you recommend an alternate approach to commit and push my changes without being blocked by these errors?  

Thank you for your support!  

Best regards
<img width="1440" alt="Screenshot 2024-12-12 at 16 37 31" src="https://github.com/user-attachments/assets/c2f9529c-3c28-4509-9ab0-a4237587521c" />

**Comment by belloibrahv** [comment]
@quiet-node @natanasow @Nana-EC Please kindly help address the issues and questions on the task.

**Comment by quiet-node** [comment]
Thank you for the update, @belloibrahv, and I apologize for my late response! Looking into these updates and will share my thoughts on them soon.

**Comment by quiet-node** [comment]
> Hi @quiet-node @natanasow,
> 
> I hope this message finds you well. I’m currently working on the issue #3142 (`Fix all TypeScript warnings and errors generated by type mismatching on the ConfigService.get(<envName>) method in test specs`).
> 
> While committing and signing off my changes, I encountered several linter errors during the build-and-test process. These errors seem unrelated to the code changes I’ve made, as they primarily point to existing issues in the repository. Below are a few examples of the errors:
> 
> * **Error**: `'accounts' is never reassigned. Use 'const' instead` (`prefer-const`)
> * **Warning**: `Require statement not part of import statement` (`@typescript-eslint/no-var-requires`)
> * **Warning**: `'data' is defined but never used` (`@typescript-eslint/no-unused-vars`)
> 
> These issues are originating from files such as:
> 
> * `/tests/integration/server.spec.ts`
> * `/tests/acceptance/getTransactionByHash.spec.ts`
> * `/tests/acceptance/sendRawTransaction.spec.ts`, and others.
> 
> I’ve ensured that my changes align with the task requirements, and the issues above were not introduced by my code.
> 
> Could you kindly guide me on how to proceed? Should I address these linter issues as part of this task, or would you recommend an alternate approach to commit and push my changes without being blocked by these errors?
> 
> Thank you for your support!
> 
> Best regards <img alt="Screenshot 2024-12-12 at 16 37 31" width="1440" src="https://private-user-images.githubusercontent.com/94493245/395229714-c2f9529c-3c28-4509-9ab0-a4237587521c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ1NDEzNTUsIm5iZiI6MTczNDU0MTA1NSwicGF0aCI6Ii85NDQ5MzI0NS8zOTUyMjk3MTQtYzJmOTUyOWMtM2MyOC00NTA5LTlhYjAtYTQyMzc1ODc1MjFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE4VDE2NTczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlNzY0N2IzNDczZTZkY2Q0NGY2ZTBiYTgzMDdkNGY5ZWNlODI4MDE2OWUzNDdhZjFiZDZhOWMyYjBjMjNkMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fHO1savEt8uju3ECOmt28vJxo2TcEXNMxyi7XW5V4F8">

Hmm, that's strange! Were you able to rebase the changes? Did the linting errors persist after the rebase? 

Our goal is to keep PRs focused and address only the reported issues for clarity. If you believe these errors were not introduced by your code, please open a new ticket in the repository to report them for easier tracking. The team will review the ticket and determine how to address them.

Thanks for the great work and effort as always!

**Comment by natanasow** [comment]
> > Hi @quiet-node @natanasow, I noticed the 'NON_EXISTING_VAR' does not exist in the `GlobalConfig.ENTRIES` as part of the `envName` listed. I suggest adding it to ensure type safety when retrieving configuration values. Or do what you think about it. <img alt="Screenshot 2024-12-12 at 14 38 41" width="904" src="https://private-user-images.githubusercontent.com/94493245/395182577-0b774c4a-e683-4539-9109-78001621b351.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ1NDEzNTUsIm5iZiI6MTczNDU0MTA1NSwicGF0aCI6Ii85NDQ5MzI0NS8zOTUxODI1NzctMGI3NzRjNGEtZTY4My00NTM5LTkxMDktNzgwMDE2MjFiMzUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE4VDE2NTczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwNGRlODFkYTBlMTJmMDkwNTdmMDAyNGM5YTk4MzA3YWE3MGMyOTYwNDE0YWYxMWM1MmM0MmUzZTU1OTYzNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HfqqGPRAELVzylNgQ158YpOPBKR-TRKMMNzd0x0hoG0">
> 
> This is a good idea! @natanasow what do you think?

@belloibrahv In that test, we want to ensure if the var is missing, the config service will not brick or throw an error. If we add the `NON_EXISTING_VAR` to the config array, we will change the test initial goal.

Let's create a new ticket for the linter errors and not commit several issues to a single PR/branch.

Thank you for the great contribution, looking forward to the PR review :rocket: .

**Comment by belloibrahv** [comment]
@Nana-EC @ebadiere @quiet-node I've submitted a PR addressing issue #3142 to resolve type warnings for ConfigService.get(). It ensures type consistency across test cases by refining type usage and adding/removing the necessary casting. Let me know if further changes are needed. Here's a in the PR https://github.com/hashgraph/hedera-json-rpc-relay/pull/3350 Thank you for reviewing!
